jwcrypto/jwe.py

```diff
@@ -485,6 +488,130 @@ def unwrap(self, key, keylen, ek, headers):
         return cek
 
 
+class _EcdhEs(_RawKeyMgmt):
+
+    @property
+    def name(self):
+        return 'ECDH-ES'
+
+    def __init__(self, keydatalen=None):
+        self.backend = default_backend()
+        self.keydatalen = keydatalen
+
+    def _check_key(self, key):
+        if key.key_type != 'EC':
+            raise InvalidJWEKeyType('EC', key.key_type)
+
+    def _derive(self, privkey, pubkey, algID, keydatalen, headers):
+        # OtherInfo is defined in NIST SP 56A 5.8.1.2.1
+
+        # AlgorithmID
+        otherinfo = struct.pack('>I', len(algID))
+        otherinfo += bytes(algID.encode('utf8'))
+
+        # PartyUInfo
+        if 'apu' in headers:
+            apu = base64url_decode(headers['apu'])
+        else:
+            apu = b''
+        otherinfo += struct.pack('>I', len(apu))
```

Comment:
I think this is cleaner:
`apu = base64url_decode(headers['apu']) if 'apu' in headers else b''`